packages/rulesets/src/spectral/test/latest-version-of-common-types-must-be-used.test.ts

```diff
@@ -0,0 +1,157 @@
+import { Spectral } from "@stoplight/spectral-core"
+import linterForRule from "./utils"
+
+let linter: Spectral
+
+beforeAll(async () => {
+  linter = await linterForRule("LatestVersionOfCommonTypesMustBeUsed")
+  return linter
+})
+
+test("LatestVersionOfCommonTypesMustBeUsed should find errors for obsolete version", () => {
+  const myOpenApiDocument = {
+    swagger: "2.0",
+    paths: {
+      "/foo": {
+        get: {
+          operationId: "foo_post",
+          parameters: [
+            {
+              $ref: "../../../../../common-types/resource-management/v3/customermanagedkeys.json#/parameters/ApiVersionParameter",
+            },
+            {
+              $ref: "../../../../../common-types/resource-management/v1/managedidentity.json#/parameters/ApiVersionParameter",
+            },
+          ],
+          responses: {
+            200: {
+              description: "Success",
+              $ref: "../../../../../common-types/resource-management/v4/types.json#/parameters/ApiVersionParameter",
+            },
+          },
+        },
+      },
+    },
+    parameters: {
+      LoadTestNameParameter: {
+        in: "path",
+        name: "loadTestName",
+        description: "Load Test name.",
+        required: true,
+        "x-ms-parameter-location": "method",
+        type: "string",
+      },
+      QuotaBucketNameParameter: {
+        in: "query",
+        name: "quotaBucketName",
+        description: "Quota Bucket name.",
+        required: true,
+        "x-ms-parameter-location": "method",
+        type: "string",
+      },
+    },
+  }
+  return linter.run(myOpenApiDocument).then((results) => {
+    const desiredResults = results.filter((result) => result.message.includes("Use the latest version"))
```

Comment:
Why does `results` need to be filtered here?

Reply:
For some reason ENT errors are getting captured as well

Reply:
Might be due how the swagger is formatted

Reply:
Ah! its cause "../../../../../common-types/resource-management/v1/managedidentity.json#/parameters/ApiVersionParameter" path doesnt exist in real

Reply:
Oh, that makes sense. If you change the linter instantiation in your test it might fix it: `linterForRule("LatestVersionOfCommonTypesMustBeUsed", true)` (added `true`). I added the option to that function because I ran into the same problem.